RecoMuon/MuonIdentification/python/earlyMuons_cfi.py

```diff
@@ -17,7 +17,8 @@
     TrackAssociatorParameters = dict(
 	useHO   = False,
 	useEcal = False,
-	useHcal = False)
+	useHcal = False),
+    pvInputTag = "firstStepPrimaryVertices" #Not used, but needed to avoid circular dependency in schedule
```

Comment:
this comment raises the question if `desc.ifValue` should perhaps be used to provide a default `pvInputTag` in case `selectHighPurity==False`, so a fake/unused value would not have to be entered.

Reply:
I looked at some examples using desc.ifValue, but I'm not getting the behavior I want.  selectHighPurity is being set to True, when not present.  Can you make a specific recommendation?

Reply:
maybe I missed the difficulty here, but why not this:
```diff
-  desc.add<bool>("selectHighPurity", false);
+  desc.ifValue(
+      edm::ParameterDescription<bool>("selectHighPurity", false, true),
+      true >> (edm::ParameterDescription<edm::InputTag>("pvInputTag", edm::InputTag("offlinePrimaryVertices"), true)) or
+      false >> (edm::ParameterDescription<edm::InputTag>("pvInputTag", edm::InputTag(""), true))
+   );
```

Reply:
The issue is that when I look interactively with python, I see that for 'earlyMuons', 'selectHighPurity' is set to True
The intended behavior was to set it to False, as accomplished by:
 desc.add<bool>("selectHighPurity", false);

Reply:
I'm not sure I follow...
```
$ python3 -c "from RecoMuon.MuonIdentification.earlyMuons_cfi import earlyMuons;print(earlyMuons.selectHighPurity)"
cms.bool(False)
```

adding @cms-sw/muon-pog-l2 in case they have any inputs. On a related note, I wonder why `muons1stStep` is implemented using an explicit duplication of parameters, rather than cloning the defaults?

Reply:
Interesting.  I confirm your observation, but from
 runTheMatrix.py -l 159 -j0
python3 -i step3_RAW2DIGI_L1Reco_RECO_EI_PAT_VALIDATION_DQM.py
I see:
process.earlyMuons.selectHighPurity
cms.bool(True)

Reply:
This seems related to the import order (`earlyMuons` is cloned from `muons1stStep` after the `pp_on_AA` is applied) and is also the case for the vanilla version of this PR.

Reply:
Indeed, and looking again, it's unrelated to the suggestion change.  I moved the era customization to RecoMuon.Configuration.RecoMuon_cff, which gives the desired behavior.